### PR TITLE
feat(3000): Support server-side search of insights and groups

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

Server-side search was not supported for insights and groups in the sidebar, which meant you could not _really_ search for them.

## Changes

That's now supported. Also, abstracted away debouncing for sidebars where items are searched server-side.

### CODE REVIEW

File: frontend/src/layout/navigation-3000/sidebars/personsAndGroupsSidebarLogic.ts [hunk 1]
```diff
@@ -59,7 +51,7 @@ export const personsAndGroupsSidebarLogic = kea<personsAndGroupsSidebarLogicType
                             } as BasicListItem
                         }),
                         loading: personsLoading,
-                        loadMore: persons.next ? () => actions.loadPersons(persons.next) : undefined,
+                        loadMore: persons.next ? () => actions.loadPersons(persons.next) : undefined, // FIXME
```
Comment (thmsobrmlr) [inline frontend/src/layout/navigation-3000/sidebars/personsAndGroupsSidebarLogic.ts:54]: fix what exactly 😅

File: frontend/src/layout/navigation-3000/sidebars/utils.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { SearchMatch } from '../types'
  * This provides highlighting for server-side search, which does not return data on how the search term was matched.
  */
 export function findSearchTermInItemName(name: string, searchTerm: string): SearchMatch | null {
-    if (!searchTerm) {
+    if (!searchTerm || !name) {
```
Comment (thmsobrmlr) [inline frontend/src/layout/navigation-3000/sidebars/utils.ts:8]: There's a stray log in line 16.

**Review by thmsobrmlr — APPROVED** [review]
Nice! Do you plan to move the search server side for other things e.g. dashboards as well?

File: frontend/src/layout/navigation-3000/sidebars/personsAndGroupsSidebarLogic.ts (hunk 1 above)
Comment (Twixes) [reply frontend/src/layout/navigation-3000/sidebars/personsAndGroupsSidebarLogic.ts:54]: Well, `loadMore()` doesn't actually do anything right now – but I've already got a PR to fix this, so cleaned up

**Comment by Twixes** [comment]
We will definitely need to move to that for dashboards, since it takes a good few seconds to load all of them for us now… So that may come, but definitely later.

**Pull request merged**